src/overlays/actors/ovl_En_Skb/z_en_skb.c

```diff
@@ -66,55 +74,499 @@ static ColliderJntSphInit D_80AFE068 = {
+    /* Kokiri spin   */ DMG_ENTRY(1 , 0xD),
+    /* Giant spin    */ DMG_ENTRY(4 , 0xF),
+    /* Master spin   */ DMG_ENTRY(2 , 0xF),
+    /* Kokiri jump   */ DMG_ENTRY(2 , 0xF),
+    /* Giant jump    */ DMG_ENTRY(8 , 0xF),
+    /* Master jump   */ DMG_ENTRY(4 , 0xF),
+    /* Unknown 1     */ DMG_ENTRY(0 , 0x0),
+    /* Unblockable   */ DMG_ENTRY(0 , 0x0),
+    /* Hammer jump   */ DMG_ENTRY(4 , 0xF),
+    /* Unknown 2     */ DMG_ENTRY(0 , 0x0),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD880.s")
+const ActorInit En_Skb_InitVars = {
+    ACTOR_EN_SKB,
+    ACTORCAT_ENEMY,
+    FLAGS,
+    OBJECT_SKB,
+    sizeof(EnSkb),
+    (ActorFunc)EnSkb_Init,
+    (ActorFunc)EnSkb_Destroy,
+    (ActorFunc)EnSkb_Update,
+    (ActorFunc)EnSkb_Draw,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFD968.s")
+Vec3f D_80AFE0B8 = { 0.0f, 8.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/EnSkb_Update.s")
+Vec3f D_80AFE0C4 = { 0.0f, -1.5f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Skb/func_80AFDD30.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(targetArrowOffset, 2000, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -2000, ICHAIN_STOP),
+};
 
+void EnSkb_SetupAction(EnSkb* this, EnSkbActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnSkb_SpawnDebris(GlobalContext* globalCtx, EnSkb* this, Vec3f* spawnPos) {
+    Vec3f pos;
+    Vec3f vel = D_80AFE0B8;
+    Vec3f accel = D_80AFE0C4;
+    f32 spreadAngle;
+    f32 scale;
+
+    spreadAngle = (Rand_ZeroOne() - 0.5f) * 6.28f;
+    pos.y = this->actor.floorHeight;
+    pos.x = (Math_SinF(spreadAngle) * 15.0f) + spawnPos->x;
+    pos.z = (Math_CosF(spreadAngle) * 15.0f) + spawnPos->z;
+    accel.x = Rand_CenteredFloat(1.0f);
+    accel.z = Rand_CenteredFloat(1.0f);
+    vel.y += ((Rand_ZeroOne() - 0.5f) * 4.0f);
+    scale = (Rand_ZeroOne() * 5.0f) + 12.0f;
+    EffectSsHahen_Spawn(globalCtx, &pos, &vel, &accel, 2, (scale * 0.8f), -1, 10, 0);
+    func_80033480(globalCtx, &pos, 10.0f, 1, 150, 0, 1);
+}
+
+void EnSkb_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSkb* this = THIS;
+    s16 paramOffsetBody;
+    s16 paramOffsetArm;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    this->actor.colChkInfo.damageTable = &D_80AFE078;
+    ActorShape_Init(&this->actor.shape, 0.0f, &ActorShadow_DrawCircle, 0.0f);
+    this->actor.focus.pos = this->actor.world.pos;
+    this->actor.colChkInfo.mass = 0xFE;
+    this->actor.colChkInfo.health = 2;
+    this->actor.shape.yOffset = -8000.0f;
+    SkelAnime_Init(globalCtx, &this->skelAnime, &D_060041F8, &D_06001854, this->limbDrawTable,
+                   this->transitionDrawTable, 20);
+    this->actor.naviEnemyId = 0x55;
+
+    Collider_InitJntSph(globalCtx, &this->collider);
+    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &D_80AFE068, this->colliderItem);
```

Comment:
```suggestion
    Collider_SetJntSph(globalCtx, &this->collider, &this->actor, &sJntSphInit , this->colliderItem);
```
Also dont forget to rename the collider init above and make it `static`